src/main/java/com/artipie/ResponseMetricsSlice.java

```diff
@@ -51,16 +48,25 @@ public ResponseMetricsSlice(final Slice origin, final Metrics metrics) {
     }
 
     @Override
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
     public Response response(
         final String rqline,
         final Iterable<Map.Entry<String, String>> rqheaders,
         final Publisher<ByteBuffer> rqbody) {
-        return connection -> this.origin.response(rqline, rqheaders, rqbody).send(
-            (rsstatus, rsheaders, rsbody) -> {
-                this.report(rqline, rqheaders, rsstatus);
-                return connection.accept(rsstatus, rsheaders, rsbody);
+        return connection -> {
+            try {
+                return this.origin.response(rqline, rqheaders, rqbody).send(
+                    (rsstatus, rsheaders, rsbody) -> {
+                        this.report(rqline, rqheaders, rsstatus);
+                        return connection.accept(rsstatus, rsheaders, rsbody);
+                    }
+                );
+            // @checkstyle IllegalCatchCheck (1 line)
+            } catch (final Exception exc) {
```

Comment:
We need to catch `ArtipieHttpException` first, since it can include some error status and message, an generic exception only after that:
```java
} catch (ArtipieHttpException ex) {
  // report with ex.status
} catch (Exception ex) {
  // report INTERNAL_ERROR
}
```

Reply:
Ok